code/modules/mob/mob_defines.dm

```diff
@@ -158,6 +158,7 @@
 
 	var/can_pull_size = 10              // Maximum w_class the mob can pull.
 	var/can_pull_mobs = MOB_PULL_LARGER // Whether or not the mob can pull other mobs.
+	var/can_climb = 1 //Whether the mob can climb ladders
```

Comment:
Don't we already have a method for this in multiz/movement? Specifically `can_z_travel` and specific ladder usage procs. Modify those. This variable should not be necessary. At all.